clouddrift/adapters/hurdat2.py

```diff
@@ -0,0 +1,452 @@
+
+_DEFAULT_FILE_PATH = os.path.join(tempfile.gettempdir(), "clouddrift", _DEFAULT_NAME)
+os.makedirs(_DEFAULT_FILE_PATH, exist_ok=True)
+
+
+class BasinOption(int, enum.Enum):
+    ATLANTIC = 1
+    NORTHEAST_PACIFIC = 2
+    BOTH = 3
+
+
+class RecordIdentifier(str, enum.Enum):
+    """
+    C – Closest approach to a coast, not followed by a landfall
+    G – Genesis
+    I – An intensity peak in terms of both pressure and wind
+    L – Landfall (center of system crossing a coastline)
+    P – Minimum in central pressure
+    R – Provides additional detail on the intensity of the cyclone when rapid changes are underway
+    S – Change of status of the system
+    T – Provides additional detail on the track (position) of the cyclone
+    W – Maximum sustained wind speed
+    """
+
+    CLOSES_TO_COAST = "C"
+    GENESIS = "G"
+    INTENSITY_PEAK = "I"
+    LANDFALL = "L"
+    MINIMUM_CENTRAL_PRESSURE = "P"
+    RAPID_CHANGE = "R"
+    STATUS_CHANGE = "S"
+    TRACK_POSITION = "T"
+    MAX_SUSTAINED_WIND_SPEED = "W"
+    NOT_AVAILABLE = ""
+
+
+class SystemStatus(str, enum.Enum):
+    """
+    TD – Tropical cyclone of tropical depression intensity (< 34 knots)
+    TS – Tropical cyclone of tropical storm intensity (34-63 knots)
+    HU – Tropical cyclone of hurricane intensity (> 64 knots)
+    EX – Extratropical cyclone (of any intensity)
+    SD – Subtropical cyclone of subtropical depression intensity (< 34 knots)
+    SS – Subtropical cyclone of subtropical storm intensity (> 34 knots)
+    LO – A low that is neither a tropical cyclone, a subtropical cyclone, nor an extratropical cyclone (of any intensity)
+    WV – Tropical Wave (of any intensity)
+    DB – Disturbance (of any intensity)
+    ET - UNKNOWN found in Northeast Pacific Basin
+    PT - UNKNOWN found in Northeast Pacific Basin
+    ST - UNKNOWN found in Northeast Pacific Basin
+    TY - UNKNOWN found in Northeast Pacific Basin
+    """
+
+    TD = "TD"
+    TS = "TS"
+    HU = "HU"
+    EX = "EX"
+    SD = "SD"
+    SS = "SS"
+    LO = "LO"
+    WV = "WV"
+    DB = "DB"
+    ET = "ET"
+    PT = "PT"
+    ST = "ST"
+    TY = "TY"
+
+
+@dataclass
+class HeaderLine:
+    basin: str = field(
+        metadata={
+            "long_name": "Basin",
+            "comments": "basin of origin.",
+        }
+    )
+    cyclone_number: int = field(
+        metadata={"long_name": "ATCF cyclone number for associated year."}
+    )
+    year: int = field(metadata={"long_name": "Year"})
+    name: str = field(metadata={"long_name": "Name", "na_values": "UNNAMED"})
+    rowsize: int = field(metadata={"long_name": "Number of best track entries"})
+    id: str = field(default_factory=lambda: uuid4().hex)
+
+
+@dataclass
+class DataLine:
+    time: float = field(
+        metadata={"comments": "Computed property from YYY-MM-DD HH:MM in UTC"}
+    )
+    record_identifier: RecordIdentifier = field(
+        metadata={"long_name": "Record Idenfier", "comments": RecordIdentifier.__doc__}
```

Comment:
Using .__doc__ introduces an extra change of line.

```
In [5]: ds.record_identifier
Out[5]:
<xarray.DataArray 'record_identifier' (obs: 30238)> Size: 121kB
array(['R', 'R', 'R', ..., 'R', 'R', 'R'], dtype='<U1')
Coordinates:
    time     (obs) float64 242kB -6.488e+08 -6.488e+08 ... 1.665e+09 1.665e+09
Dimensions without coordinates: obs
Attributes:
    long_name:  Record Idenfier
    comments:   \n    C – Closest approach to a coast, not followed by a land...
```

Reply:
addressed this in new commits